modules/build/src/main/scala/scala/build/preprocessing/mdsandbox/SnippetPackager.scala

```diff
@@ -0,0 +1,147 @@
+package scala.build.preprocessing.mdsandbox
+
+import MarkdownSnippet.Fence
+
+class SnippetPackager(val fileName: String, val snippets: Seq[Fence]) {
+  def this(fileName: String, content: String) = this(fileName, MarkdownSnippet.findFences(content))
+
+  val (rawSnippets, processedSnippets) = snippets.partition(f => f.isRaw)
+  val (testSnippets, runSnippets) = processedSnippets.partition(f => f.isTest)
+
+  val runObjectIdentifier: String = s"`Markdown$$${fileName.replace('.', '$')}`"
+  val testObjectIdentifier: String = s"`Markdown_Test$$${fileName.replace('.', '$')}`"
+
+  /** Generates class name for a snippet with given index */
+  def runClassName(index: Int): String = s"Snippet$$$index"
+
+  /** Generates class name for a test snippet with given index */
+  def testClassName(index: Int): String = s"`Test$$${fileName.replace('.', '$')}_$index`"
+
+  /** Returns Scala snippets packed into classes and glued together into an object */
+  def buildScalaMain(): String = {
+    if (runSnippets.isEmpty) s"object $runObjectIdentifier {def execute(): Unit = {}}"  // no snippets
+    else (0 until runSnippets.length).foldLeft(
+      s"object $runObjectIdentifier {def execute(): Unit = {"
+    ) (
+      (sum, index) => 
+        if (runSnippets(index).resetScope || index == 0) sum :++ s"new ${runClassName(index)}; "
+        else sum  // that class hasn't been created
```

Comment:
?

Reply:
If we have following snippets:
- `scala`
- `scala`
- `scala reset`

the wrapper will create `Snippet0` and `Snippet2` classes for them. The middle snippet, as its scope has not been reset, does not have its own class, and so `Snippet1` class is being omitted here not to try to execute a non-existent class (thus raising an exception)